src/pani/core.cljs

```diff
@@ -0,0 +1,161 @@
+(ns pani.core
+  (:refer-clojure :exclude [get-in set! reset! conj! swap! deref parents key])
+  (:require cljsjs.firebase
+            [clojure.string :as str]
+            [pani.registry :refer [register-listener]]))
+
+;; constants
+
+(def child-events
+  [:child-added
+   :child-changed
+   :child-moved
+   :child-removed])
+
+(def all-events
+  (conj child-events :value))
+
+;; utils
+
+;; FIXME: camel-case keys?
+;;        hydrate to custom vectors to preserve rich keys?
+;;        preserve sets (don't coerce to vector)
+;;        similarly preserve keywords as values
+
+(defn- hydrate [js-val]
+  (js->clj js-val :keywordize-keys true))
+
+(defn- serialize [clj-val]
+  (clj->js clj-val))
```

Comment:
Yeah, you'd pull back a map. We could coerce to (sorted-map) to at least keep the lexical ordering stable, but then we're losing priorities set on the ref. This is a bigger problem once we support custom ordering queries.

For this reason I think we should detect (suspiciously sized keys)[https://www.firebase.com/blog/2015-02-11-firebase-unique-identifiers.html] and hydrate to some kind a vector / map hybrid (ideal world), that allows say reordering, filtering via seq functions, and stable rewriting. Idealist :)